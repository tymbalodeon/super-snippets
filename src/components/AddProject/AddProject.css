.AddProject {
  text-align: center;
  margin-top: 50px;
}

.AddProject * > button {
  margin-top: 20px;
  margin-left: 5px;
  margin-right: 5px;
  padding: 5px;
}

.AddProject > h2 {
  color: #5ccfe6;
}

.AddProject * > button[type='submit'] {
  background-color: #ffa759;
  border-color: #ffa759;
}

.back {
  background-color: #cbccc6;
}
