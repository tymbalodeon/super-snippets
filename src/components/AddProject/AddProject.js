import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import SnippetApiService from '../../services/snippet-api-service.js';
import './AddProject.css';

export default class AddProject extends Component {
  handleSubmit = e => {
    e.preventDefault();
    const { title } = e.target;
    const { addProject } = this.props;
    SnippetApiService.postProject(title.value)
      .then(addProject)
      .then(() => {
        title.value = '';
      })
      .catch(error => this.setState({ error }));
  };

  render() {
    return (
      <section className="AddProject">
        <h2>Create a project</h2>
        <form onSubmit={this.handleSubmit}>
          <div className="field">
            <label htmlFor="project-name-input">Name: </label>
            <input type="text" id="project-name-input" name="title" required />
          </div>
          <div className="buttons">
            <Link to="#" onClick={this.props.history.goBack}>
              <button className="back">BACK</button>
            </Link>
            <button type="submit">ADD</button>
          </div>
        </form>
      </section>
    );
  }
}
