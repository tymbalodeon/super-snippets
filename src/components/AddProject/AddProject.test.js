import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import AddProject from './AddProject';

describe('<AddProject />', () => {
  const history = createMemoryHistory('/');

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <AddProject history={history} />
      </BrowserRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AddProject history={history} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
