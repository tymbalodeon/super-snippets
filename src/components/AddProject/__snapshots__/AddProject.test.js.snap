// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddProject /> renders the UI as expected 1`] = `
<section
  className="AddProject"
>
  <h2>
    Create a project
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="project-name-input"
      >
        Name: 
      </label>
      <input
        id="project-name-input"
        name="title"
        required={true}
        type="text"
      />
    </div>
    <div
      className="buttons"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="back"
        >
          BACK
        </button>
      </a>
      <button
        type="submit"
      >
        ADD
      </button>
    </div>
  </form>
</section>
`;
